### PR TITLE
11.0 mig website_logo

### PR DESCRIPTION
**Pull request by ernestotejeda**
cc @Tecnativa

### CODE REVIEW

File: website_logo/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    'depends': [
+        'website',
+    ],
+    'data': [
+        'views/res_config_view.xml',
+        'views/website_view.xml',
+        'views/website_templates.xml',
+    ],
+    'qweb': [
+    ],
```
Comment (MeetKD) [inline website_logo/__manifest__.py]: Avoid empty key.

File: website_logo/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    'name': 'Website logo',
+    'summary': 'Website company logo',
+    'version': '11.0.1.0.0',
+    'category': 'Website',
+    'author': "Agile Business Group, "
+              "LasLabs, "
+              "Tecnativa, "
+              "Odoo Community Association (OCA)",
+    'website': 'http://www.agilebg.com',
```
Comment (MeetKD) [inline website_logo/__manifest__.py]: Replace website with oca website URL.

**Review by MeetKD — CHANGES_REQUESTED** [review]

File: website_logo/models/res_config.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import models, fields
+
+
+class WebsiteConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    logo = fields.Binary(
+        string="Website logo", related="website_id.logo",
+        help="This field holds the logo for this website, showed in header."
```
Comment (simahawk) [inline website_logo/models/res_config.py]: no need to repeat params on related field

File: website_logo/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        if row and row[0]:
+            return io.BytesIO(base64.b64decode(row[0])), row[1]
+        return False, False
+
+    @http.route([
+        '/web/binary/website_logo',
+        '/website_logo',
+        '/website_logo.png',
+    ], type='http', auth="none", cors="*")
+    def website_logo(self, dbname=None, **kw):
```
Comment (simahawk) [inline website_logo/controllers/main.py:36]: question: do we really need this whole machinery just to get the logo?
Why can't we simply get current website from request and grab logo value?

File: website_logo/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        imgname = 'website_nologo.png'
```
Comment (simahawk) [inline website_logo/controllers/main.py:37]: nologo image seems empty, am I wrong? Should we have a `no logo` image or the default odoo one?

File: website_logo/views/res_config_view.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    <field name="arch" type="xml">
+        <xpath expr="//div[@id='domain_setting']//div[hasclass('text-muted')][1]" position="attributes">
+            <attribute name="invisible">1</attribute>
+        </xpath>
+        <xpath expr="//div[@id='domain_setting']//div[hasclass('text-muted')][1]" position="after">
+            <div class="text-muted">
+                Name, favicon and logo of your website
+            </div>
+        </xpath>
+        <xpath expr="//div[@id='domain_setting']//field[@name='favicon']/.." position="after">
```
Comment (simahawk) [inline website_logo/views/res_config_view.xml:18]: shouldn't we state that `png` format is required?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by rafaelbn** [comment]
cc @Tecnativa

File: website_logo/views/res_config_view.xml (hunk 6 above)
Comment (pedrobaeza) [reply website_logo/views/res_config_view.xml:18]: That's not requiredf AFAIK

File: website_logo/controllers/main.py (hunk 4 above)
Comment (ernestotejeda) [reply website_logo/controllers/main.py:36]: We can use current website method and obtain the current website, but this method, if it not find a website that has the current domain, return the firs website un database. But, with the current code, if it not find a website tha has the current domain, it try to find a website that has the 'localhost' domain. That is a differece. If it doesn't matter, we can use the method.

File: website_logo/controllers/main.py (hunk 5 above)
Comment (ernestotejeda) [reply website_logo/controllers/main.py:37]: It is the text placed in Usage section of README.rst, it was written before the migration:

This addon supports multi-website. If no logo is defined, a transparent image is
placed, allowing to have no logo in one website, even if "Show logo" option is
set.

**Comment by JGarcia-Panach** [comment]
Tested funcionally 👍

**Review by JGarcia-Panach — APPROVED** [review]
Functional review 👍

**Review by rafaelbn — APPROVED** [review]
Functionaly tested :+1: 

Please @chienandalu @simahawk @MeetKD could you make here last review in technical staff ? Thank you!

File: website_logo/__init__.py [hunk 7]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/__init__.py]: :fire:

File: website_logo/controllers/__init__.py [hunk 8]
```diff
@@ -1,3 +1,2 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/controllers/__init__.py]: :fire:

File: website_logo/controllers/main.py [hunk 9]
```diff
@@ -1,8 +1,8 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/controllers/main.py]: :fire:

File: website_logo/models/__init__.py [hunk 10]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/models/__init__.py]: :fire:

File: website_logo/models/website.py [hunk 11]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/models/website.py]: :fire:

File: website_logo/tests/__init__.py [hunk 12]
```diff
@@ -1,3 +1,2 @@
-# -*- coding: utf-8 -*-
```
Comment (yajo) [inline website_logo/tests/__init__.py]: :fire:

File: website_logo/tests/test_logo.py [hunk 13]
```diff
@@ -22,7 +22,8 @@ def setUp(self):
         self.model_obj = self.env['res.company']
         self.cont_obj = Website()
         self.rec_id = self.env.ref('base.main_company')
-        self.image_val = 'Test'.encode('base64')
+        # self.image_val = 'Test'.encode('base64')
```
Comment (yajo) [inline website_logo/tests/test_logo.py]: :fire:

File: website_logo/views/res_config_view.xml [hunk 14]
```diff
@@ -1,19 +1,27 @@
...
+            <div class="row">
+                <label class="col-md-3 o_light_label" for="logo" string="Logo"/>
+                <field name="logo" widget="image"/>
             </div>
         </xpath>
+
```
Comment (yajo) [inline website_logo/views/res_config_view.xml]: :fire:

**Review by yajo — APPROVED** [review]
Lol, I expected to see something more, but code is OK besides some unneeded whitespace and comments. It would be great if you can clean those.

**Review by simahawk — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@Yajo is a pyromaniacal :stuck_out_tongue_closed_eyes:

**Pull request merged**